### DISCUSSION TITLE
Draw models as lines instead of fill - wireframe like

### DISCUSSION THREAD

**Discussion by rrocav3** [sceneview/sceneform-android#147] [Q&A]
Is there a way to show lines that conform the faces instead of the model faces? 

SceneKit like:
`node.geometry?.firstMaterial?.fillMode = .lines`

OpenGl like:
`glPolygonMode(GL_FRONT_AND_BACK, GL_LINE)`

Expected result:
![image](https://user-images.githubusercontent.com/42677484/131157644-826063d4-af34-43e0-a86f-b6ae4b87ef07.png)

Thanks!

**Comment by ThomasGorisse** [comment]
Hi,

~~It sounds possible within Filament and Sceneform gives you access to the FilamentAsset directly.~~
**Filament only provides a wireframe representation for the bounding box.**

So this Filament PR: https://github.com/google/filament/pull/942 by @prideout ~~should~~**should not** match your need.
You can then get the **bounding box** **Wireframe Renderable** from your **gltf object**.

The only thing missing is the java link to this cpp function inside `FilamentAsset.java`:
```cpp
/**
 * Lazily creates a single LINES renderable that draws the transformed bounding-box hierarchy
 * for diagnostic purposes. The wireframe is owned by the asset so clients should not delete it.
*/
utils::Entity getWireframe() noexcept;
```

~~Could you make a little quick PR on the Filament repo and link it here?~~

**Reply by ThomasGorisse** [reply]
Sorry, I just saw the function doc and it seems to only get the Bounding box wireframe.

@prideout can you confirm that there is no actual way to get the object wireframe?

**Reply by prideout** [reply]
Right, that's just a bounding box.  Filament's core API does not support glPolygonMode, that might be a feature request you could make over on the Filament github page.  (This would be an actual core Filament feature, not a gltfio feature.)

**Reply by rrocav3** [reply]
Hello @prideout I currently open a thread in filament for asking this https://github.com/google/filament/discussions/4558
Romain said I have to handle it for myself by using a custom material and I try to follow a guide with no success. I think opengl can handle it for itself as long as I find other samples rendering wireframes like https://github.com/the3deers/android-3D-model-viewer

Thanks in advance!

**Reply by prideout** [reply]
Apologies, Romain is correct, Filament is mobile first and it must support OpenGL ES class devices.  Therefore the wireframe needs to be rendered using a method other than glPolygonMode.

**Reply by rrocav3** [reply]
Sure, I see glPolygonMode is not in OpenGL ES but thought this can be done by filament using another method.
